### PR TITLE
CloudForms OpenShift networking solution document

### PR DESCRIPTION
**Pull request by sabre1041**
Document created to outline the network solution for integrating OpenShift and CloudForms 

@etsauer @oybed @JaredBurck

### CODE REVIEW

File: playbooks/operationalizing/cloudforms_networking.adoc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+Since CloudForms requires a single point of entry for container providers, a Local Traffic Manager can be utilized to create a VIP backed by a set of virtual servers and corresponding load balancing pools representing components within OpenShift to control the flow of traffic. A VIP contains a single DNS that contains two virtual servers in the LTM. A virtual server is represented by an IP address and a service (port in this case) that will receive traffic for further processing. These include:
+
+* ose.example.com listening on port 8443 to process traffic destined for the OpenShift API
+* ose.example.com listening on port 5000 to process traffic destined for the OpenShift Metrics components (Hawkular)
+
+Also configured in the LTM are two sets of load balancing pools. Load balancing pools are logical sets of devices that are grouped together to receive and process traffic. These include:
+
+* The addresses of each OpenShift master instance to pass traffic on port 8443 for the API
+* THe addresses of each OpenShift node where each integrated HAProxy router is running on port 443.
```
Comment (etsauer) [inline playbooks/operationalizing/cloudforms_networking.adoc]: minor capitalization issue

File: playbooks/operationalizing/cloudforms_networking.adoc [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+* The addresses of each OpenShift master instance to pass traffic on port 8443 for the API
+* THe addresses of each OpenShift node where each integrated HAProxy router is running on port 443. 
+
+Once these two components have been defined, the virtual server associated to port 8443 can include the load balancing pool containing the set of master instances on port 8443. The virtual server associated to port 5000 can include the load balancing pool containing the set of node instance that contain running instances of the integrated HAProxy router to pass traffic to the Hawkular metrics services within the OpenShift cluster.
+
+
+==== SSL Considerations
+
+To provide the greatest level of flexibility, SSL can be configured in several places depending on the architecture of the environment. SSL termination for the API can occur at eiter the LTM or OpenShift level. For the metrics component, SSL termination can either occur at the LTM layer, Hawkular layer or at the HAProxy layer with the help of SSL reencryption support. The following links provide examples of how each type are configured:
```
Comment (etsauer) [inline playbooks/operationalizing/cloudforms_networking.adoc]: It seems like somewhere in here (or perhaps elswhere in this doc) there is an opportunity to reference theplaybooks Load Balancing Guide.

File: playbooks/operationalizing/cloudforms_networking.adoc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+A response similar to the following should be returned:
+
+[source]
+----
+NAME         HOST/PORT            PATH      SERVICE            LABELS    INSECURE POLICY   TLS TERMINATION
+cloudforms   ose.example.com             hawkular-metrics                               passthrough
+----
+
+Verify Hawkular can be reached by navigating to the following location in a web browser: https://ose.example.com:5000
```
Comment (etsauer) [inline playbooks/operationalizing/cloudforms_networking.adoc]: I think we need a closing line at the end about now moving on to normal CloudForms integration. And eventually we will point them back to the main CF doc once published.

**Comment by etsauer** [comment]
@sabre1041 overall looks pretty good. Just some minor stuff. Also I think there are a lot of opportunities to link out to other playbooks. In general I want us to be better about that, and maybe we need a style guide addition to reflect inter-linking.

**Comment by sabre1041** [comment]
@etsauer PR has been updated to account for your suggestions

File: playbooks/operationalizing/cloudforms_networking.adoc [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+* When master instances are also configured as nodes, they are marked as unschedulable by default. This ensures no applications will be deployed on these instances, which could potentially affect available resources and segregate the performance of the master 
+* Additional network firewall ports to open on master instances
+
+The purpose of this document is to provide an alternative solution for integrating CloudForms and OpenShift by leveraging a Local Traffic Manager (LTM), such as F5 Big-IP(™), to provide the mechanisms of routing traffic based on service type to the underlying components.
+
+== Solution Overview
+
+To overcome the requirement of deploying a router on OpenShift master instances, the provisioning and configuration of a VIP using an LTM alongside configurations in the OpenShift environment will be required. 
+
+NOTE: It is reccommended that the link:../installation/load_balancing.adoc[External Load Balancers Guide] which covers the load balancing strategies in OpenShift be reviewed prior to implementing this solution.
```
Comment (etsauer) [inline playbooks/operationalizing/cloudforms_networking.adoc]: please use absolute paths for playbook links

**Comment by sabre1041** [comment]
@etsauer removed the relative path link

**Comment by etsauer** [comment]
lgtm. @oybed @JaredBurck can I get a second?

File: playbooks/operationalizing/cloudforms_networking.adoc [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+These resources are provided by two separate sets of services in the OpenShift environment: the API and cluster metrics. The API is a service that runs on the OpenShift master instances to provide access the topology of the environment, while cluster metrics are provided as running application containers from the link:http://www.hawkular.org/[Hawkular] project to measure the performance of the environment. In CloudForms today, the Container provider communicates with a single fixed hostname or IP address. This becomes a challenge when integrating CloudForms with OpenShift as both types of services must be accessed from a single host. Most architectures of OpenShift split out master and application (serviced by router pods running on nodes) communication in order to provide a separation of concerns and to optimize network traffic. To overcome this limitation, it is recommended in the link:https://access.redhat.com/documentation/en/red-hat-cloudforms/version-4.0/managing-providers/[CloudForms Provider Management documentation] that a secondary application router be deployed on each master instance with the sole purpose of serving service traffic for CloudForms.
```
Comment (JayKayy) [inline playbooks/operationalizing/cloudforms_networking.adoc]: 2nd sentence is missing a 'to'. "...provide access *to* the topology of the environment"

File: playbooks/operationalizing/cloudforms_networking.adoc [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+* When master instances are also configured as nodes, they are marked as unschedulable by default. This ensures no applications will be deployed on these instances, which could potentially affect available resources and segregate the performance of the master 
+* Additional network firewall ports to open on master instances
+
+The purpose of this document is to provide an alternative solution for integrating CloudForms and OpenShift by leveraging a Local Traffic Manager (LTM), such as F5 Big-IP(™), to provide the mechanisms of routing traffic based on service type to the underlying components.
+
+== Solution Overview
+
+To overcome the requirement of deploying a router on OpenShift master instances, the provisioning and configuration of a VIP using an LTM alongside configurations in the OpenShift environment will be required. 
+
+NOTE: It is reccommended that the link:/playbooks/installation/load_balancing.adoc[External Load Balancers Guide] which covers the load balancing strategies in OpenShift be reviewed prior to implementing this solution.
```
Comment (JayKayy) [inline playbooks/operationalizing/cloudforms_networking.adoc]: Spelling: Recommended

**Comment by JayKayy** [comment]
Looks pretty good to me. I put in a few annoyingly insignificant changes. see comments.

**Comment by sabre1041** [comment]
@JayKayy corrections have been made

**Comment by JayKayy** [comment]
I'm good with this.

**Pull request merged**